store/secretsmgr_test.go

```diff
@@ -0,0 +1,496 @@
+package store
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager/types"
+	"github.com/aws/smithy-go/middleware"
+	"github.com/guardian/devx-config/log"
+	"regexp"
+	"testing"
+	"time"
+)
+
+type SecretsManagerClientMock struct {
```

Comment:
Caveat subjective, but mocks are quite often confusing and not a helpful testing strategy - as they are very coupled to details and so brittle. In any case, I'm not sure that these tests are adding value here - see comment below.

Reply:
https://martinfowler.com/articles/mocksArentStubs.html is an interesting (and balanced) article on testing styles anyway. As it says, there are definitely cases where behavioural testing is helpful (e.g. a cache). But on the whole, I'd favour what it calls the 'classic' approach.